drivers/led_strip/ws2812_i2s.c

```diff
@@ -0,0 +1,255 @@
+                    break;
+                case LED_COLOR_ID_BLUE:
+                    pixel = pixels[i].b;
+                    break;
+                default:
+                    return -EINVAL;
+            }
+            ws2812_i2s_ser(tx_buf, pixel, sym_one, sym_zero);
+            tx_buf++;
+        }
+    }
+
+    for(uint16_t i = 0; i < cfg->reset_words; i++) {
+        *tx_buf = reset_word;
+        tx_buf++;
+    }
+
+    /* Flush the buffer on the wire. */
+    ret = i2s_write(cfg->dev, mem_block, cfg->tx_buf_bytes);
+    if (ret < 0) {
+        k_mem_slab_free(cfg->mem_slab, &mem_block);
+        LOG_ERR("Failed to write data: %d", ret);
+        return ret;
+    }
+
+    ret = i2s_trigger(cfg->dev, I2S_DIR_TX, I2S_TRIGGER_START);
+    if (ret < 0) {
+        LOG_ERR("Failed to trigger command %d on TX: %d", I2S_TRIGGER_START, ret);
+        return ret;
+    }
+
+    ret = i2s_trigger(cfg->dev, I2S_DIR_TX, I2S_TRIGGER_DRAIN);
+    if (ret < 0) {
+        LOG_ERR("Failed to trigger command %d on TX: %d", I2S_TRIGGER_DRAIN, ret);
+        return ret;
+    }
+
+    /* Wait until transaction is over */
+    flush_time_us = cfg->tx_buf_bytes / (sizeof(uint32_t) * 8);
+    flush_time_us *= 10;        /* 1 LRCK period = 1 32b word */
+    k_usleep(flush_time_us);
+
+    return ret;
+}
+
+static int ws2812_strip_update_channels(const struct device *dev,
+                                        uint8_t *channels,
+                                        size_t num_channels)
+{
+    LOG_ERR("update_channels not implemented");
+    return -ENOTSUP;
+}
+
+static int ws2812_i2s_init(const struct device *dev)
+{
+    const struct ws2812_i2s_cfg *cfg = dev->config;
+    uint8_t i;
+    int ret;
+
+    struct i2s_config config;
+
+    /* 16-bit stereo, 100kHz LCLK */
+    config.word_size = 16;
+    config.channels = 2;
+    config.format = I2S_FMT_DATA_FORMAT_I2S;
+    config.options = I2S_OPT_BIT_CLK_MASTER | I2S_OPT_FRAME_CLK_MASTER;
+    config.frame_clk_freq = WS2812_I2S_LRCK_HZ; /* WS (or LRCK) */
+    config.mem_slab = cfg->mem_slab;
+    config.block_size = cfg->tx_buf_bytes;
+    config.timeout = 1000;
+
+    ret = i2s_configure(cfg->dev, I2S_DIR_TX, &config);
+    if (ret < 0) {
+        LOG_ERR("Failed to configure I2S device: %d\n", ret);
+        return ret;
+    }
+
+    for (i = 0; i < cfg->num_colors; i++) {
+        switch (cfg->color_mapping[i]) {
+            case LED_COLOR_ID_WHITE:
+            case LED_COLOR_ID_RED:
+            case LED_COLOR_ID_GREEN:
+            case LED_COLOR_ID_BLUE:
+                break;
+            default:
+                LOG_ERR("%s: invalid channel to color mapping."
+                        "Check the color-mapping DT property",
+                        dev->name);
+                return -EINVAL;
+        }
+    }
+
+    return 0;
+}
+
+static const struct led_strip_driver_api ws2812_i2s_api = {
+.update_rgb = ws2812_strip_update_rgb,
+.update_channels = ws2812_strip_update_channels,
```

Comment:
Some tabs are missing above.

Please run checkpatch.pl against the driver and fixes the whole indentation. Tabulations are expanded into 4 spaces everywhere...

Reply:
yep sorry about that, I haven't yet installed the editorconfig plugin on my personal machine, I'll fix the compliance issues.